tidy3d/components/data/monitor_data.py

```diff
@@ -133,7 +133,136 @@ def symmetry_expanded_copy(self) -> AbstractFieldData:
+        return plane_bounds0, plane_bounds1
+
+    @cached_property
+    def _diff_area(self) -> xr.DataArray:
+        """For a 2D monitor data, return the area of each cell in the plane, for use in numerical
+        integrations."""
+        bounds = self._plane_grid_boundaries
+        sizes = [bs[1:] - bs[:-1] for bs in bounds]
+        return xr.DataArray(np.outer(sizes[0], sizes[1]), dims=self._tangential_dims)
+
+    @property
+    def _centered_tangential_fields(self) -> Dict[str, DataArray]:
+        """For a 2D monitor data, get the tangential E and H fields colocated to the cell centers in
+        the 2D plane grid."""
+
+        # Tangential directions and fields
+        tan_dims = self._tangential_dims
+        tan_fields = self._tangential_fields
+
+        # Plane center coordinates
+        bounds = self._plane_grid_boundaries
+        centers = [(bs[1:] + bs[:-1]) / 2 for bs in bounds]
+
+        # Interpolate tangential field components to cell centers
+        interp_dict = dict(zip(tan_dims, centers))
+        centered_fields = {key: val.interp(**interp_dict) for key, val in tan_fields.items()}
+        return centered_fields
+
+    @property
+    def poynting(self) -> xr.DataArray:
+        """Time-averaged Poynting vector for frequency-domain data associated to a 2D monitor."""
+
+        # Tangential fields are ordered as E1, E2, H1, H2
+        tan_fields = self._centered_tangential_fields
+        field_list = list(tan_fields.values())
+        e_x_h_star = field_list[0] * field_list[3].conj() - field_list[1] * field_list[2].conj()
+        poynting = 0.5 * np.real(e_x_h_star)
+        # cast to xr.DataArray as this is technically no longer a ScalarFieldArray
+        return xr.DataArray(poynting)
+
+    @property
+    def flux(self) -> xr.DataArray:
+        """Flux for data corresponding to a 2D monitor."""
+
+        # Compute flux by integrating Poynting vector in-plane
+        d_area = self._diff_area
+        return (self.poynting * d_area).sum(dim=d_area.dims)
+
+    def dot(
+        self, field_data: Union[FieldData, ModeSolverData], conjugate: bool = True
+    ) -> xr.DataArray:
+        """Dot product (modal overlap) with another ``field_data`` object. Both datasets have to be
+        frequency-domain data associated with a 2D monitor. Along the tangential directions,
+        the datasets have to have the same discretization. Along the normal direction, the monitor
+        position may differ and is ignored. Other coordinates (``frequency``, ``mode_index``) have
+        to be either identical or broadcastable.
+
+        Parameters
+        ----------
+        field_data : :class:`ElectromagneticFieldData`
+            A data instance to compute the dot product with.
+        conjugate : bool, optional
+            If ``True`` (default), the dot product is defined as ``1 / 4`` times the integral of
+            ``E1* x H2 - H1* x E2``, where ``x`` is the cross product, ``*`` is complex conjugation,
```

Comment:
Just a sanity check, in line 206 it seemed like the H1 and H2 are conjugated, not E1 and H1 as said here. Am I missing something? (note: I am assuming E1 is the 1st tangential component of E)

Reply:
oh ok, so in this notations 1=self and 2=the other field data? maybe this could be clarified in the line below ("E2, H2 is the supplied field_Data) or something.

Reply:
And maybe we should just find a different conversion for self and other, like E1 -> E_self

Reply:
Updated docstring.